src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -544,6 +574,64 @@ private function hasProductWithNameAndShortDescription(array $products, string $
         return false;
     }
 
+    private function hasProductOptionWithNameAndValue(string $optionValue, string $optionName): bool
+    {
+        $response = $this->client->getLastResponse();
+        $productOptions = $this->responseChecker->getValue($response, 'options');
+
+        foreach ($productOptions as $optionIri) {
+            if ($this->hasProductOptionWithName($optionIri, $optionName)) {
+                $variants = $this->responseChecker->getValue($response, 'variants');
+
+                foreach ($variants as $variantIri) {
+                    if ($this->variantHasProductOptionValue($variantIri, $optionValue))
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
```

Comment:
some styling recommendations. If you reverse first if condition, you are reducing function nesting 

```suggestion
        foreach ($productOptions as $optionIri) {
            if (!$this->hasProductOptionWithName($optionIri, $optionName)) {
                continue;
            }

            $variants = $this->responseChecker->getValue($response, 'variants');

            foreach ($variants as $variantIri) {
                if ($this->variantHasProductOptionValue($variantIri, $optionValue)) { /* you had wrong brackets here */
                    return true;
                }
            }
        }
```